tools/tool-testing/selftest.js

```diff
@@ -151,6 +151,14 @@ export function define(name, tagsList, f) {
   }));
 }
 
+// Prevent specific self-test's from being defined and run.
+// e.g. `selftest.skip.define("some test", ...` will skip running "some test".
+export const skip = {
+  define() {
+    /* no-op */
```

Comment:
Sometimes test frameworks will print something to indicate that a test is being skipped. Would that be useful here?

Reply:
Thanks for the suggestion @benjamn - I originally logged a "Skipped" message, but decided to take it out when I saw how it looked alongside the final `self-test` summary message. I'll take another pass at it though - thanks!

Reply:
Just as an example, this is how it looked. 

![screenshot 2018-01-26 20 22 41](https://user-images.githubusercontent.com/137740/35466960-0f20a17a-02d7-11e8-8490-f4e29e635d09.png)

Because `skip` is called before `define`, the message gets logged when all of the tests are first attempted to be defined. So even when using a `self-test` exclude pattern, all `.skip` configured tests are still dumped to the console. Maybe that's okay - I just thought it looked a bit odd ...

Reply:
I've added a skipped test message that looks like this:

```
...
Ignoring "old cli tests (converted)" test (excluded by selftest.skip)
Ignoring "compiler plugins - addAssets" test (excluded by selftest.skip)
...
```

Reply:
Yeah, I guess I was thinking we should only log the skips if those tests would have been run. If that's not easy, this isn't essential.